tests/platform_tests/mellanox/test_psu_power_threshold.py

```diff
@@ -0,0 +1,292 @@
+    mocker = mocker_factory(duthost, 'PsuPowerThresholdMocker')
+    if MockPlatform:
+        if all_psus_supporting_thresholds:
+            logger.info('CLI option "--mock_any_testbed" is provided while power thresholds are supported on both PSUs')
+
+        logger.info('Mocking the system to support PSU power threshold')
+        mocker.mock_power_threshold(MAX_PSUS)
+
+        # Restart PSU daemon to take the mock stuff
+        logger.info('Restart PSU daemon to take mock PSU power threshold')
+        duthost.shell('docker exec -ti pmon supervisorctl restart psud')
+        psudaemon_restarted = True
+        time.sleep(2)
+    elif not all_psus_supporting_thresholds:
+        pytest.skip('PSU power threshold is not supported')
+
+    yield
+
+    logging.info('Clean all mock files')
+    mocker.deinit()
+
+    if psudaemon_restarted:
+        logger.info('Restore PSU daemon')
+        duthost.shell('docker exec -ti pmon supervisorctl restart psud')
+        time.sleep(2)
+
+
+def init_log_analyzer(duthost, marker, expected, ignored=None):
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix=marker)
+    marker = loganalyzer.init()
+
+    loganalyzer.load_common_config()
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expected)
+    if ignored:
+        loganalyzer.ignore_regex.extend(ignored)
+
+    return loganalyzer, marker
+
+
+def check_log_analyzer(loganalyzer, marker):
+    loganalyzer.analyze(marker)
+    return loganalyzer
+
+
+@pytest.mark.disable_loganalyzer
+def test_psu_power_threshold(request, duthosts, rand_one_dut_hostname, mock_power_threshold):
+    def _check_psu_info_in_db(psu_index, power, power_warning_threshold, power_critical_threshold, power_overload):
+        psuname = 'PSU {}'.format(psu_index)
+        command_check_psu_db = 'sonic-db-cli STATE_DB hmget "PSU_INFO|{}" power power_warning_threshold power_critical_threshold power_overload'.format(psuname)
+        output = duthost.shell(command_check_psu_db)['stdout'].split()
+        if int(float(output[0])) != power/1000000 \
+               or int(float(output[1])) != power_warning_threshold/1000000 \
+               or int(float(output[2])) != power_critical_threshold/1000000 \
+               or output[3] != str(power_overload):
+            return False
+
+        command_check_system_health_db = 'sonic-db-cli STATE_DB hget SYSTEM_HEALTH_INFO "{}"'
+        summary = duthost.shell(command_check_system_health_db.format('summary'))['stdout'].strip()
+        if power_overload:
+            if 'Not OK' in summary:
+                detail = duthost.shell(command_check_system_health_db.format(psuname))['stdout'].strip()
+                if 'exceeds threshold' in detail:
+                    return True
+        elif summary == 'OK':
+            return True
+        else:
+            detail = duthost.shell(command_check_system_health_db.format(psuname))['stdout'].strip()
```

Comment:
log the detail here in case PSU failed due to other reason?

Reply:
L131